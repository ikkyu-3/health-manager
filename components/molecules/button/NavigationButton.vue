<template>
  <v-btn :value="value">
    <span>{{ text }}</span>
    <v-icon>{{ icon }}</v-icon>
  </v-btn>
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class NavigationButton extends Vue {
  @Prop(String) readonly value!: string
  @Prop(String) readonly text!: string
  @Prop(String) readonly icon!: string
}
</script>
