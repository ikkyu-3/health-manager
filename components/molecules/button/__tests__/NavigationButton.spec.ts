import { mount } from '@vue/test-utils'
import NavigationButton from '@/components/molecules/button/NavigationButton.vue'

describe('molecules/button/NavigationButton.vue', () => {
  it('snapshot', () => {
    const wrapper = mount(NavigationButton, {
      propsData: {
        value: 'workouts',
        text: 'Workouts',
        icon: 'fa-dumbbell'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
