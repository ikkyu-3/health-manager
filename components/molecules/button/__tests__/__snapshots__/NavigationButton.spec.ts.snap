// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/button/NavigationButton.vue snapshot 1`] = `"<button type=\\"button\\" value=\\"workouts\\" class=\\"v-btn v-btn--contained theme--light v-size--default\\"><span class=\\"v-btn__content\\"><span>Workouts</span> <i aria-hidden=\\"true\\" class=\\"v-icon notranslate fa fa-dumbbell theme--light\\"></i></span></button>"`;
